api/src/v2/Controller/ClientController.php

```diff
@@ -61,7 +61,7 @@ public function getByIdAction(int $id): JsonResponse
 
     /**
      * @Route("/case-number/{caseNumber}", methods={"GET"})
-     * @Security("has_role('ROLE_ADMIN') or has_role('ROLE_AD')")
+     * @Security("has_role('ROLE_ADMIN') or has_role('ROLE_AD') or has_role('ROLE_DEPUTY')")
```

Comment:
I think we may need to tighten up the security on this to protect against exposing client info accidentally to people who shouldn't see it. As we limited the use of the endpoint to admins only prior to this change there may be instances where we were relying on the security rule. Could we get some kind of check on if the logged-in user is not an admin and not associated with the client then we throw a 403 or similar?

Reply:
Added this check now